src/Authentication/Helpers/EventlogHelper.cs

```diff
@@ -17,22 +23,114 @@ public static class EventlogHelper
+        {            
+            if (await featureManager.IsEnabledAsync(FeatureFlags.AuditLog))
+            {
+                AuthenticationEvent authenticationEvent = MapAuthenticationEvent(jwtToken, eventType);              
+                eventLog.CreateAuthenticationEvent(authenticationEvent);
+            }
+        }
+
+        /// <summary>
+        /// Creates an authentication event
+        /// </summary>
+        /// <param name="featureManager">handler for feature manager service</param>
+        /// <param name="eventLog">handler for eventlog service</param>
+        /// <param name="authenticatedUser">authenticat</param>
+        /// <param name="eventType">authentication event type</param>
+        public async static Task CreateAuthenticationEvent(IFeatureManager featureManager, IEventLog eventLog, UserAuthenticationModel authenticatedUser, AuthenticationEventType eventType)
         {
-            if (eventType != null)
+            if (await featureManager.IsEnabledAsync(FeatureFlags.AuditLog))
             {
-                userAuthentication.EventType = eventType;
+                AuthenticationEvent authenticationEvent = MapAuthenticationEvent(authenticatedUser, eventType);
+                eventLog.CreateAuthenticationEvent(authenticationEvent);
             }
-            else
+        }
+
+        /// <summary>
+        /// Maps claims to the authentication event model
+        /// </summary>
+        /// <param name="jwtToken">authenticated token</param>
+        /// <param name="eventType">authentication event type</param>
+        /// <returns>authentication event</returns>
+        public static AuthenticationEvent MapAuthenticationEvent(string jwtToken, AuthenticationEventType eventType)
+        {
+            JwtSecurityTokenHandler tokenHandler = new JwtSecurityTokenHandler();
+            AuthenticationEvent authenticationEvent = null;
+            if (!string.IsNullOrEmpty(jwtToken))
             {
-                userAuthentication.EventType = userAuthentication.IsAuthenticated ? AuthenticationEventType.Authenticated : AuthenticationEventType.AuthenticationFailed;
+                JwtSecurityToken token = tokenHandler.ReadJwtToken(jwtToken);
+
+                if (token != null)
+                {
+                    authenticationEvent = new AuthenticationEvent();
+                    foreach (Claim claim in token.Claims)
+                    {
+                        // Handle various claim types
+                        switch (claim.Type)
+                        {
+                            case AltinnCoreClaimTypes.UserId:
+                                authenticationEvent.UserId = claim.Value;
+                                break;
+
+                            case AltinnCoreClaimTypes.Org:
+                                authenticationEvent.OrgNumber = claim.Value;
```

Comment:
Is it by itention you use same property for org and orgnumber. 90290324 would overwrite "SKD"

Reply:
No, I was not aware that claim type org comes with "SKD". I underdstood that climtype org is a internal notation and claimtype orgnumber was another type of token. I will fix it